### DISCUSSION TITLE
About document improvement

### DISCUSSION THREAD

**Discussion by script-money** [seanjameshan/starknet.js#883] [General]
When using starknetjs to complete CTF challenges, I often call lower-level functions and have encountered several issues with the documentation (as I am using JS instead of TS):
1. The documentation lacks a search function.
2. Searching for specific functions in the GitHub source code results in multiple versions of the documentation, making it difficult to locate the specific function.
3. Many functions lack comments and explanations. For example, the "version" in different functions is referred to as cairoVersion or transactionVersion, but both are only labeled as "version" without any comments. To find the specific transaction version, I needs to navigate to src/utils/hash.ts -> src/constants.ts to find correct value.
4. The guide contains outdated content and context is discontinuous and not suitable for beginners.

Suggestions:
1. Start a new documentation library using a framework with a search function.
2. Add function and type explanations as much as possible.
3. Rewrite or update the guide.

**Comment by ivpavici** [comment]
hi!
Search will be added in the near future: https://github.com/starknet-io/starknet.js/issues/875

**Reply by ivpavici** [reply]
Regarding the rest of the points... the Straknet ecosystem is evolving rapidly and unfortunately things like docs and guides sometimes lag with changes. But they will catch up

**Comment by Maksone34334** [comment]
Hi! Good Project!
